File: src/main/java/com/yerti/core/database/SQLiteManager.java
package com.yerti.core.database;

import org.bukkit.plugin.Plugin;

import java.io.File;
import java.sql.*;

public class SQLiteManager {

    File file;
    private Connection connection;

    public SQLiteManager(Plugin plugin, String database, String user, String pass) {

        try {
            Class.forName("org.sqlite.JDBC");

            file = new File(plugin.getDataFolder(), database + ".sqlite");

            if (!file.getParentFile().exists()) {
                file.getParentFile().mkdir();
            }

        } catch (Exception e) {
            e.printStackTrace();
        }

    }

    public void execute(String param) {
        try {
            this.connection = DriverManager.getConnection("jdbc:sqlite:" + file.getAbsolutePath());
            Statement statement = connection.createStatement();
            statement.execute(param);
            statement.close();
            this.connection.close();

        } catch (SQLException e) {
            e.printStackTrace();
        }
    }

    public void executeUpdate(String param) {
        try {
            this.connection = DriverManager.getConnection("jdbc:sqlite:" + file.getAbsolutePath());
            Statement statement = connection.createStatement();
            statement.executeUpdate(param);
            statement.close();
            this.connection.close();

        } catch (SQLException e) {
            e.printStackTrace();
        }
    }

    public ResultSet query(String param) {
        try {
            this.connection = DriverManager.getConnection("jdbc:sqlite:" + file.getAbsolutePath());
            Statement statement = connection.createStatement();
            ResultSet set = statement.executeQuery(param);
            statement.close();
            this.connection.close();
            return set;

        } catch (SQLException e) {
            e.printStackTrace();
        }

        return null;
    }

}

File: src/main/java/com/yerti/core/database/SQLiteManager.kt
package com.yerti.core.database

import org.bukkit.plugin.Plugin
import java.io.File
import java.sql.Connection
import java.sql.DriverManager
import java.sql.ResultSet
import java.sql.SQLException

class SQLiteManager(plugin: Plugin, database: String, user: String?, pass: String?) {
    private var connection: Connection? = null
    var file: File? = null
    fun execute(param: String?) {
        try {
            connection = DriverManager.getConnection("jdbc:sqlite:" + file!!.absolutePath)
            val statement = connection!!.createStatement()
            statement.execute(param)
            statement.close()
            connection!!.close()
        } catch (e: SQLException) {
            e.printStackTrace()
        }
    }

    fun executeUpdate(param: String?) {
        try {
            connection = DriverManager.getConnection("jdbc:sqlite:" + file!!.absolutePath)
            val statement = connection!!.createStatement()
            statement.executeUpdate(param)
            statement.close()
            connection!!.close()
        } catch (e: SQLException) {
            e.printStackTrace()
        }
    }

    fun query(param: String?): ResultSet? {
        try {
            connection = DriverManager.getConnection("jdbc:sqlite:" + file!!.absolutePath)
            val statement = connection!!.createStatement()
            val set = statement.executeQuery(param)
            statement.close()
            connection!!.close()
            return set
        } catch (e: SQLException) {
            e.printStackTrace()
        }
        return null
    }

    init {
        try {
            Class.forName("org.sqlite.JDBC")
            file = File(plugin.dataFolder, "$database.sqlite")
            if (!file!!.parentFile.exists()) {
                file!!.parentFile.mkdir()
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}